pandas/core/internals/managers.py

```diff
@@ -1905,7 +1905,9 @@ def _merge_blocks(
     return blocks
 
 
-def _compare_or_regex_search(a, b, regex=False):
+def _compare_or_regex_search(
+    a: Union[ArrayLike, Scalar], b: Union[ArrayLike, Scalar], regex: bool = False
```

Comment:
This typing of "ArrayLike" is not necessarily correct. Although the name might indicate otherwise, ArrayLike is specifically ndarray+ExtensionArray, and not general array-like

(to be clear, it might also be correct, I didn't check where this function is called)

Reply:
I think in this case the ArrayLike is already covering the right use cases for the function.